### DISCUSSION TITLE
How to add new translation file for new language

### DISCUSSION THREAD

**Discussion by mrmmg** [DesignPatternsPHP/DesignPatternsPHP#475] [Q&A]
Hi

I have decided to participate in open source projects.
My native language is Persian (Farsi) and usually the documents that novice programmers need are written at high levels of English language that are difficult for them to read.

For this reason, I decided to provide a suitable Persian translation for this repository.  Please advise me on how to translate English to Persian, I mean how can I find English language files with the po extension?

**Comment by domnikl** [comment]
Hi @mrmmg,
that's a great idea! You can copy one of the folders in https://github.com/DesignPatternsPHP/DesignPatternsPHP/tree/main/locale and rename it to the ISO code for Persian (Farsi). Every file contains the english phrases in the `msgid` fields, so you can delete the existing translations and add your own instead. After your PR is merged, I will then add it to ReadTheDocs and it will be linked to the English version.

**Reply by mrmmg** [reply]
Hi @domnikl 
Ok, very nice
I will work on translation :)
